src/Models/AZ3166Device.ts

```diff
@@ -188,46 +188,8 @@ export class AZ3166Device extends ArduinoDeviceBase {
       let deviceConnectionString =
           ConfigHandler.get<string>(ConfigKey.iotHubDeviceConnectionString);
 
-      let hostName = '';
-      let deviceId = '';
-      if (deviceConnectionString) {
-        const hostnameMatches =
-            deviceConnectionString.match(/HostName=(.*?)(;|$)/);
-        if (hostnameMatches) {
-          hostName = hostnameMatches[0];
-        }
-
-        const deviceIDMatches =
-            deviceConnectionString.match(/DeviceId=(.*?)(;|$)/);
-        if (deviceIDMatches) {
-          deviceId = deviceIDMatches[0];
-        }
-      }
-
-      let deviceConnectionStringSelection: vscode.QuickPickItem[] = [];
-      if (deviceId && hostName) {
-        deviceConnectionStringSelection = [
-          {
-            label: 'Select IoT Hub Device Connection String',
-            description: '',
-            detail: `Device Information: ${hostName} ${deviceId}`
-          },
-          {
-            label: 'Input IoT Hub Device Connection String',
-            description: '',
-            detail: ''
-          }
-        ];
-      } else {
-        deviceConnectionStringSelection = [{
-          label: 'Input IoT Hub Device Connection String',
-          description: '',
-          detail: ''
-        }];
-      }
-
       const selection = await vscode.window.showQuickPick(
-          deviceConnectionStringSelection,
+          this.getDeviceConnectionStringOptions(deviceConnectionString),

```

Comment:
>this.getDeviceConnectionStringOptions(deviceConnectionString) [](start = 10, length = 61)

Use a variable 'deviceConnectionStringSelection' to store the result of this function and pass the variable as parameter to showQuickPick() function for readability.